**Câu 151:** Viết chương trình kiểm tra tin nhắn có chữ thường hay không nếu có trả về True ngược lại trả về False.

In [1]:
message = 'Toilaai2'
for msg in message:
    if msg.islower():
        print(True)
        break
else:
    print(False)

True


**Câu 152:** Viết chương trình lọc các số lẻ từ danh sách được người dùng nhập vào các số cách nhau bởi dấu phẩy.

<div style="padding: 15px; border: 2px solid transparent; border-color: transparent; margin-bottom: 5px; margin-top: 10px;border-radius: 2px; color: #000000; background-color:#A4D3EE; border-color: #FFD39B;Width:60%">

Example:
<br>
<br>
Input: 1,2,3,4,5,6,7,8,9,10
<br>
<br>
Output: 1,3,5,7,9

</div>

In [2]:
values=input("Nhập các số cách nhau bởi đấu phẩy:").split(",")
results=[]
for v in values:
    if int(v)%2!=0:
        results=results+[v]
print("Các số lẻ là",",".join(results))

Nhập các số cách nhau bởi đấu phẩy:1,2,3,4,5,6,7,8,9,10
Các số lẻ là 1,3,5,7,9


**Câu 153:** Viết hàm tính số cặp các số nguyên giống nhau của một mảng.
<div style="padding: 15px; border: 2px solid transparent; border-color: transparent; margin-bottom: 5px; margin-top: 10px;border-radius: 2px; color: #000000; background-color:#A4D3EE; border-color: #FFD39B;Width:60%">

Example:
<br>
<br> 
Input: [10, 20, 20, 10, 10, 30, 50, 10, 20]
<br>
<br>
Output: 3

</div>

In [3]:
def sockMerchant(ar):
    set_ar=set(ar)
    counter=0
    for s in set_ar:
        counter+=ar.count(s)//2
    return counter
sockMerchant([10, 20, 20, 10, 10, 30, 50, 10, 20])

3

**Câu 154:** Hiện thực hàm stringlen() trả lại chiều dài của chuỗi.

In [4]:
def stringlen(A):
    count=0
    for i in A:
        count+=1
    return count
stringlen("aame")

4

**Câu 155:** Viết hàm xác định phần tử đầu tiên có giá trị là số nguyên tố và nhỏ hơn 100 và trả về vị trí của phần tử đó, ngược lại trả về -1.

In [5]:
def Prime(n):
    if n<=2:
        return False
    else:
        for i in range(2,n):
            if n%i==0:
                return False
    return True
def FindPrimeArray(A):
    for i in range(len(A)):
        if A[i]<100:
            if Prime(A[i]):
                return i
        else: 
            return -1
FindPrimeArray([1,3,5,19])

1

**Câu 156:** Viết chương trình xáo trộn chuỗi.

In [6]:
import random
char_list = ['a','e','i','o','u']
random.shuffle(char_list)
print(''.join(char_list))

ieauo


**Câu 157:** Viết chương trình tính tiền điện dựa vào số Kw tiêu thụ biết khung giá tiền điện như sau:
+ Từ 0kW đến 100kW giá 500d/kW
+ Từ 100kW trở lên  đến 250kW giá 800d/kW
+ Từ 250kW trở lên đến 350kW giá 1000d/kW
+ Từ 350kW trở lên giá 1500d/kW 

In [7]:
values=float(input("Nhập số kW tiên thụ trong tháng:"))
money=values*500
if values>100 and values>=250:
    money=money+(values-100)*800
if values>250 and values<=350:
    money=money+(values-250)*1000
if values>350:
    money=money+(values-350)*1500
print("Giá tiền điện tháng này là:{} d".format(money))

Nhập số kW tiên thụ trong tháng:423
Giá tiền điện tháng này là:579400.0 d


**Câu 158:** Viết chương trình nhận chuỗi đầu vào từ giao diện điều khiển và in nó theo thứ tự ngược lại.

<div style="padding: 15px; border: 2px solid transparent; border-color: transparent; margin-bottom: 5px; margin-top: 10px;border-radius: 2px; color: #000000; background-color:#A4D3EE; border-color: #FFD39B;Width:60%">

Example:
<br>
<br>
Input: anh
<br>
<br>
Output: hna
</div>


In [8]:
chuoi=input("Nhập chuỗi vào đây: ")
chuoi = chuoi[::-1]
print (chuoi)

Nhập chuỗi vào đây: anh
hna


**Câu 159:** Viết hàm để in ra tất cả các số nguyên tố trong một phạm vi nhất định $m<= x <=n$.
<div style="padding: 15px; border: 2px solid transparent; border-color: transparent; margin-bottom: 5px; margin-top: 10px;border-radius: 2px; color: #000000; background-color:#A4D3EE; border-color: #FFD39B;Width:60%">

Example:
<br>
<br>
Input: printprime(17,25)
<br>
<br>
Output: 17 19 23 
</div>

In [9]:
def printprime(m,n):
    for i in range(m,n+1):
        count=0
        for j in range(1,i+1):
            
            if i%j==0:
                count+=1
        if count==2:
            print(j,end=' ')
    print("\nĐã kiểm tra")
printprime(17,25)   

17 19 23 
Đã kiểm tra


**Câu 160:** Một hàm getRooms yêu cầu đầu vào là tên và nếu người dùng có tên bắt đầu bằng chữ cái giữa P và S, với các thông báo sau hiển thị đến người dùng:
- Nếu đúng khóa: "Bạn đi đến phòng 223 để lấy khóa"
- Khác: "Bạn có thể lấy khóa tại đây"

Lưu ý: không phân biệt ký tự hoa và thường.

In [10]:
def getRooms(string):
    if string[0]=='P' or string[0]=='S':
        print("Bạn đi đến phòng 223 để lấy khóa")
    else:
        print("Bạn có thể lấy khóa tại đây")
getRooms("Phon")

Bạn đi đến phòng 223 để lấy khóa


**Câu 161:** Xác định số lượng phần tử có giá trị lẻ trong mảng 1 chiều.

In [11]:
def Odd_NumArray(A):
    d=0
    for i in range(len(A)):
        if A[i]%2!=0:
            d+=1
    print(d)
Odd_NumArray([3,4,5,7,8])

3


**Câu 162:** Viết chương trình nhập vào một số nhị phân ở dạng chuỗi ký tự 0 và 1, chuyển số này thành một số nguyên hệ thập phân.

In [12]:
values=input("Nhập vào số nhị phân bất kỳ:")
results=0
for i in range(len(values),0,-1):
    results=results+int(values[len(values)-i])*2**(len(values)-i)
print(results)

Nhập vào số nhị phân bất kỳ:10100
5


**Câu 163:** Một Robot di chuyển trong mặt phẳng bắt đầu từ điểm đầu tiên (0,0). Robot có thể di chuyển theo hướng UP, DOWN, LEFT và RIGHT với những bước nhất định. Dấu di chuyển của robot được đánh hiển thị như sau:

UP 5

DOWN 3

LEFT 3

RIGHT 3

Các con số sau phía sau hướng di chuyển chính là số bước đi. Hãy viết chương trình để tính toán khoảng cách từ vị trí hiện tại đến vị trí đầu tiên, sau khi robot đã di chuyển một quãng đường. Nếu khoảng cách là một số thập phân chỉ cần in số nguyên gần nhất.


<div style="padding: 15px; border: 2px solid transparent; border-color: transparent; margin-bottom: 5px; margin-top: 10px;border-radius: 2px; color: #000000; background-color:#A4D3EE; border-color: #FFD39B;Width:60%">

Example:
<br>
<br>
Input: printprime(17,25)
<br>
<br>
UP 5
<br>
<br>
DOWN 3
<br>
<br>
LEFT 3
<br>
<br>
RIGHT 2
<br>
<br>
Output:
<br>
<br>
DISTANCE = 2
</div>




In [13]:
import math 
pos = [0,0]
while True:
    s = input()
    if not s:
        break
    move = s.split(" ")
    direct = move[0]
    steps = int(move[1])
    if direct=="UP":
        pos[0]+=steps
    elif direct=="DOWN":
        pos[0]-=steps
    elif direct=="LEFT":
        pos[1]-=steps
    elif direct=="RIGHT":
        pos[1]+=steps
    else:
        pass
distance=math.sqrt(pos[1]**2+pos[0]**2)
print ("DISTANCE= ",int(round(distance)))

DOWN 3
UP 5
LEFT 3
RIGHT 2

DISTANCE=  2


**Câu 164:** Định nghĩa một hàm có tạo ra dictionary chứa các key là số từ 1 đến 20 và giá trị values là bình phương của chúng. Hãy in ra giá trị values của dictionary đó.

In [14]:
def printDict():
    d=dict() 
    for i in range(1,21): 
         d[i]=i**2 
    for (k,v) in d.items(): 
           print (v,end=' ') 

printDict()

1 4 9 16 25 36 49 64 81 100 121 144 169 196 225 256 289 324 361 400 

**Câu 165:** Định nghĩa hàm có thể nhận hai số nguyên trong dạng chuỗi và tính tổng của chúng, sau đó in tổng ra giao diện điều khiển.

In [15]:
def printValue(s1,s2):
    print (int(s1)+int(s2))
printValue("3","4")

7


**Câu 166:** Viết chương trình nhận chuỗi do người dùng nhập vào và in các ký tự có chỉ số chẵn.

In [16]:
msg=input("Nhập chuỗi vào đây: ")
msg_new = msg[::2]
print (msg_new)

Nhập chuỗi vào đây: 32me
3m


**Câu 167:** Viết chương trình cộng 2 chuỗi nhị phân được nhập từ bàn phím.

In [17]:
values_1=input("Nhập chuỗi nhị phân thứ 1:")
values_2=input("Nhập chuỗi nhị phân thứ 2:")
decimal_1=int(values_1,2)
decimal_2=int(values_2,2)
Sum=bin(decimal_1+decimal_2)
print("Tổng hai số nhị phân là",Sum[2:])# Bỏ đi ký hiệu 0b

Nhập chuỗi nhị phân thứ 1:1010
Nhập chuỗi nhị phân thứ 2:1000
Tổng hai số nhị phân là 10010


**Câu 168:** Viết hàm xác định phần tử đầu tiên có giá trị lớn nhất trong mảng 1 chiều.

In [18]:
def Max_Array(A):
    Max=0
    for i in A:
        if i>Max:
            Max=i
    print(Max)
Max_Array([3,4,12,9])

12


**Câu 169:** Viết chương trình nhập một chuỗi và một ký tự. Tìm tất cả các vị trí xuất hiện của ký tự key='c' trong một chuỗi string.

In [19]:
seq=input("Nhập vào một chuỗi bất kỳ:")
results=[]
for i in range(len(seq)):
    if seq[i]=='c':
        results+=[str(i)]
print("Key 'c' nằm ở vị trí index thứ: {}".format(','.join(results)))


Nhập vào một chuỗi bất kỳ:Ban co biet trai cam khong
Key 'c' nằm ở vị trí index thứ: 4,17


**Câu 170:** Viết chương trình nhập hai chuỗi và xác định xem chúng có được tạo ra với cùng các ký tự hay không. Ví dụ “dear” và “reader” là hai chuỗi tạo từ cùng các ký tự a, d, e, r.

In [20]:
seq_1=input("Nhập chuỗi thứ 1:")
seq_2=input("Nhập chuỗi thứ 2:")
list_1=list(seq_1)
list_2=list(seq_2)
set_1=set(list_1)
set_2=set(list_2)
test=True if set_1==set_2 else False
test

Nhập chuỗi thứ 1:dear
Nhập chuỗi thứ 2:reader


True

**Câu 171:** Viết chương trình nhập hai chuỗi sau đó chèn chuỗi 2 vào chuỗi 1 tại vị trí thứ k nhập từ bàn phím.

In [21]:
seq_1=input("Nhập chuỗi thứ 1:")
seq_2=input("Nhập chuỗi thứ 2:")
k=int(input("Nhập vị trí index cần chèn vài chuỗi 1:"))
seq_new=''
for i in range(k+1):
    seq_new=seq_new+seq_1[i]
for i in range(len(seq_2)):
    seq_new=seq_new+seq_2[i]
for i in range(k+1,len(seq_1)):
    seq_new=seq_new+seq_1[i]
seq_new

Nhập chuỗi thứ 1:Toilaai
Nhập chuỗi thứ 2:ngungoc
Nhập vị trí index cần chèn vài chuỗi 1:3


'Toilngungocaai'

**Câu 172:** Viết chương trình in ra tam giác pascal.

In [22]:
from math import factorial
n = 5
for i in range(n):
    for j in range(n-i+1):
         print(end=" ")
    for j in range(i+1):
        print(factorial(i)//(factorial(j)*factorial(i-j)), end=" ")

    print()

      1 
     1 1 
    1 2 1 
   1 3 3 1 
  1 4 6 4 1 


**Câu 173:** Viết chương trình tạo ra dictionary mà có nhiều key một values.

In [23]:
mydict= {}

x, y, z = 5,3,4
mydict[x, y, z] =x+y 
 
x, y, z = 2,3,9
mydict[x, y, z] = x-z;
 
print(mydict)

{(5, 3, 4): 8, (2, 3, 9): -7}


**Câu 174:** Số tự nhiên có n chữ số là một số Armstrong (còn gọi là narcissistic numbers
hoặc pluperfect digital invariants - PPDI) nếu tổng các lũy thừa bậc n của các chữ số
của nó bằng chính nó. Hãy kiểm tra n phải là số Armstrong hay không.
<div style="padding: 15px; border: 2px solid transparent; border-color: transparent; margin-bottom: 5px; margin-top: 10px;border-radius: 2px; color: #000000; background-color:#A4D3EE; border-color: #FFD39B;Width:60%">

Example: 1^3 + 5^3 + 3^3 = 153
<br>
<br>
Input: 153
<br>
<br>
Output: 153 là số Armstrong
</div>

In [24]:
x=153
x_orgin=x
results=[]
Sum=0
while x>0:
    temp=x%10
    x=int(x/10)
    results.append(temp)
for r in results:
    Sum+=r**3
if Sum==x_orgin:
    print("{} là số Armstrong".format(x_orgin))
else:
    print("{} không là số Armstrong".format(x_orgin))

153 là số Armstrong


**Câu 175:** Hãy sử dụng module datetime in ra thời gian thực hiện tại.

In [25]:
from datetime import datetime

time = datetime.now().time() 
  
print("Bây giờ là:", time)

Bây giờ là: 09:05:20.826353


**Câu 176:** Hãy dùng hàm getsizeof trong module sys kiểm tra tập hợp A, B, C chiếm bao nhiêu dung lượng lưu trữ.

In [26]:
import sys
set_A={1,3,4,5}
set_B={"a","b","c","d"}
set_C={1,"a",3,"c",4,"d"}
print("Kích thước của tập A là:",sys.getsizeof(set_A),"byte")
print("Kích thước của tập B là:",sys.getsizeof(set_B),"byte")
print("Kích thước của tập C là:",sys.getsizeof(set_C),"byte")

Kích thước của tập A là: 116 byte
Kích thước của tập B là: 116 byte
Kích thước của tập C là: 372 byte


**Câu 177:** Viết chương trình để đọc chuỗi ASCII và chuyển đổi nó sang một chuỗi Unicode được mã hóa bằng UTF-8.

In [27]:
s = input()
v = s.encode()
print (v)

bà
b'b\xc3\xa0'


**Câu 178:** Viết chương trình tính: f(n) = f(n-1)+100 khi n>0 và f(0) = 1, với n là số được nhập vào (n>0).
<div style="padding: 15px; border: 2px solid transparent; border-color: transparent; margin-bottom: 5px; margin-top: 10px;border-radius: 2px; color: #000000; background-color:#A4D3EE; border-color: #FFD39B;Width:60%">

Example:
<br>
<br>
Input: 5
<br>
<br>
Output: 500

</div>



In [28]:
def f(n):
    if n==0:
        return 0
    else:
        return f(n-1)+100

n=int(input("Nhập số n>0: "))
print (f(n))

Nhập số n>0: 5
500


**Câu 179:** Viết hàm kiểm tra  list A có là tập chứa của list B hay không?

In [29]:
def contains(A,B):
    set_A=set(A)
    set_B=set(B)
    results=True if set_A&set_B else  False
    return results
A=[1,3,4,5]
B=[4,5,3,5,9]
contains(A,B)

True

**Câu 180:** Viết chương trình tính tổng của các values trong một dictionary.

In [30]:
def Sum_values(dic):
    Sum=0
    for (k,v) in dic.items():
        Sum+=v
    return Sum
dict_A={"a":100,"b":200,"c":200}
Sum_values(dict_A)

500

**Câu 181:** Viết một chương trình tính giá trị của a+aa+aaa+aaaa bằng cách ép qua kiểu int với a là số được nhập vào bởi người dùng.

<div style="padding: 15px; border: 2px solid transparent; border-color: transparent; margin-bottom: 5px; margin-top: 10px;border-radius: 2px; color: #000000; background-color:#A4D3EE; border-color: #FFD39B;Width:60%">

Example:
<br>
<br>
Input: n = 1
<br>
<br>
Output: 1234

</div>


In [31]:
a = input("Nhập số a: ")
n1 = int( "%s" % a )
n2 = int( "%s%s" % (a,a) )
n3 = int( "%s%s%s" % (a,a,a) )
n4 = int( "%s%s%s%s" % (a,a,a,a) )
print ("Tổng cần tính là: ",n1+n2+n3+n4)# 4+44+444+4444

Nhập số a: 4
Tổng cần tính là:  4936


**Câu 182:** Viết chương trình loại bỏ ký tự thứ 3 trong một chuỗi.

In [32]:
seq='Banlaai'
results= ""
for i in range(len(seq)):
    if i != 3:
        results = results + seq[i]
print ("Chuỗi mới sau khi loại bỏ ký tự thứ 3 trong chuỗi là:" + results)

Chuỗi mới sau khi loại bỏ ký tự thứ 3 trong chuỗi là:Banaai


**Câu 183:** Viết chương trình swap hai số a và b.

In [33]:
a=4;b=5
a,b=b,a
print("Số a là",a)
print("Số b là",b)

Số a là 5
Số b là 4


**Câu 184:** Viết chương trình để giải 1 câu đố cổ của Trung Quốc: Một trang trại thỏ và gà có 35 đầu, 94 chân, hỏi số thỏ và gà là bao nhiêu?

In [34]:
def giai(dau,chan):
    results='Không có dáp án phù hợp!'
    for i in range(dau+1):
        j=dau-i
        if 2*i+4*j==legs:
            return i,j
    return results,results

heads=35
legs=94
dap_an=giai(heads,legs)
print (dap_an)

(23, 12)


**Câu 185:** Tìm các bộ (trâu đứng, trâu nằm, trâu già) thỏa mãn bài toán cổ:

*Trăm trâu ăn trăm bó cỏ*

*Trâu đứng ăn năm*

*Trâu nằm ăn ba*

*Lụ khụ trâu già*

*Ba con một bó.*

Thử tìm cách giảm số vòng lặp khi tính toán xuống.

In [35]:
for x in range(1,int(100/5)):
    for y in range(1,int(100/3)):
        z=100-(x+y)
        if 300==15*x+9*y+z:
            print(x,y,z,sep=' ')

4 18 78
8 11 81
12 4 84


**Câu 186:**  Hiện thực hàm để tính hiệu các số chẳn trong khoảng [m,n]  với m, n >= 0.

Nếu m < n thì xuất ra "Input Error!!!"

Ngược lại xuất ra kết quả hiệu các số chẵn trong khoảng đó.
<div style="padding: 15px; border: 2px solid transparent; border-color: transparent; margin-bottom: 5px; margin-top: 10px;border-radius: 2px; color: #000000; background-color:#A4D3EE; border-color: #FFD39B;Width:60%">

Example:
<br>
<br>
Input: MinusInt(9,3)     
<br>
<br>
Output: -10

</div>

In [36]:
def MinusInt(m,n):
    if m<n:
        print("Input Error!!!")
    else:
        dem=0
        h=0
        for i in range(n,m+1):
            if i%2==0:
                dem+=1
                if dem ==1:
                    h=i
                else:
                    h-=i
        if dem==1:
            print("Hiệu các số chẵn là: ",i)
        elif dem==0:
            print("Not Values")
        else:
            print("Hiệu các số chẵn là: ",h)
MinusInt(9,3)           

Hiệu các số chẵn là:  -10


**Câu 187:** Viết chương trình chuyển đổi nhiệt độ với các tùy chọn từ người dùng
    
- Nếu người dùng nhâp vào ký tự 'F' thì cho nhập vào nhiệt độ temp và đổi sang nhiệt độ C theo công thức:

C = (5/9)(temp - 32)

- Trường hợp ký tự 'C' thì thì cho nhập vào nhiệt độ temp và đổi sang nhiệt độ F theo công thức:

F = (9/5)temp + 32

- Trường hợp khác thì xuất ra "Tuy chon khong hop le!!!"

In [37]:
option=input("Chọn option (C/F): ")
if option=="C":
    temp=float(input("Nhập vào nhiệt độ(F): "))
    print("Chuyển đổi nhiệt độ sang C: ",5/9*(temp-32))
elif option=='F':
    temp=float(input("Nhập vào nhiệt độ(C): "))
    print("Chuyển đổi nhiệt độ sang F: ",9/5*temp+32)
else:
    print("Tùy chọn không hợp lệ")

Chọn option (C/F): C
Nhập vào nhiệt độ(F): 190
Chuyển đổi nhiệt độ sang C:  87.77777777777779


**Câu 188:** Viết chương trình in list từ list [12,24,35,24,88,120,155,88,120,155], sau khi đã xóa hết các giá trị trùng nhau.

In [39]:
def delete_duplicate( listA ):
    list_new=[]
    SetA = set()
    for i in listA:
        if i not in SetA:
            SetA.add(i)
            list_new.append(i)
            
    return list_new

listA=[12,12,15,24,35,35,24,88,120,155,88,120,155]
print ("List sau khi xóa giá trị trùng là:",delete_duplicate(listA))

List sau khi xóa giá trị trùng là: [12, 15, 24, 35, 88, 120, 155]


**Câu 189:**  Viết hàm tìm kiếm nhị phân để tìm các item trong một list đã được sắp xếp. Hàm sẽ trả lại chỉ số của phần tử được tìm thấy trong list nếu không tìm thấy sẽ trả về "-1".

In [40]:
import math
def bin_search(li, element):
    bottom = 0
    top = len(li)-1
    index = -1
    while top>=bottom and index==-1:
        mid = int(math.floor((top+bottom)/2.0))
        if li[mid]==element:
            index = mid
        elif li[mid]>element:
            top = mid-1
        else:
            bottom = mid+1
    return index

li=[2,5,7,9,11,17,222]
print (bin_search(li,11))
print (bin_search(li,12))

4
-1


**Câu 190:** Viết chương tìm số nhỏ nhất trong hai số a và b và sử dụng if expression.

In [41]:
a,b=109,201
Min=b if a>=b else a 
print('Số nhỏ nhất trong {0} và {1} là {2}'.format(a,b,Min))

Số nhỏ nhất trong 109 và 201 là 109


**Câu 191:** Hãy viết hàm tính khoảng cách điểm M(x,y) đến đường thẳng ax + by + c = 0 với a, b, c được nhập từ bàn phím.

In [42]:
from math import sqrt
def distance(x,y):
    dis=abs(a*x+b*y+c*y)/sqrt(a**2+b**2)
    return dis
a=float(input("Nhập vào a: "))
b=float(input("Nhập vào b: "))
c=float(input("Nhập vào c: "))
xM=float(input("Nhập vào xM: "))
yM=float(input("Nhập vào yM: "))
distance(xM,yM)

Nhập vào a: 3
Nhập vào b: 4
Nhập vào c: 5
Nhập vào xM: 2
Nhập vào yM: 2


4.8

**Câu 192:** Viết chương trình kiểm tra một danh sách có 2 số 15 và 3 trở lên số 7 nếu đúng thì trả về True ngược lại trả về False.

In [43]:
def test(nums):
    return nums.count(15) == 2 and nums.count(7) >= 3
nums = [15,15,18,2,7,7,7,3]
test(nums)

True

**Câu 193:** Viết chương trình tạo 1 mảng nxn rỗng toàn phần tử 0.

In [44]:
n=3
matrix=[[0]*n]*n
print(matrix)

[[0, 0, 0], [0, 0, 0], [0, 0, 0]]


**Câu 194:** Tính tổng các thành phần của tuple.

In [45]:
tuple_A=(1,3,4,5)
list_A=list(tuple_A) # Hàm sum chỉ tính tổng của list được nên phải ép qua dạng list
print("Tổng các thành phần của tuple là",sum(list_A))

Tổng các thành phần của tuple là 13


**Câu 195:** Cho mảng số nguyên hãy viết hàm tìm số lớn nhất trong mỗi cặp của mảng.
<div style="padding: 15px; border: 2px solid transparent; border-color: transparent; margin-bottom: 5px; margin-top: 10px;border-radius: 2px; color: #000000; background-color:#A4D3EE; border-color: #FFD39B;Width:60%">

Example:
<br>
<br>
Input: 4 5 6 2 9 5    
<br>
<br>
Output: 5 6 6 9 9

</div>

In [46]:
def MaxAdjacent(arr,n):
    results=[]
    for i in range(1,n):
        r=max(arr[i],arr[i-1])
        results.append(r)
    for r in results:
        print(r, end=" ")
arr=[4,5,6,2,9,5]
n=6
MaxAdjacent(arr,6)

5 6 6 9 9 

**Câu 196:** Cho hàm số y = ax + b hãy tính hệ số góc và góc hợp với trục hoành. 

In [47]:
from math import atan
a=float(input("Nhập vào a: "))
b=float(input("Nhập vào b: "))
print("Hệ số góc là",a)
if a>0:
    print("Góc hợp với trục hoành là: ",atan(a)*180/3.14)
else:
    print("Góc hợp với trục hoành là: ",180-atan(-a)*180/3.14)

Nhập vào a: -3
Nhập vào b: 3
Hệ số góc là -3.0
Góc hợp với trục hoành là:  108.39864998990899


**Câu 197:** Tìm chuỗi dài nhất trong danh sách các chuỗi:

In [48]:
def test(words):
    return  max(words, key=len) 
strs =  ['cat', 'car', 'fear', 'center']
print("Chuỗi gốc:")
print(strs)
print("Chuỗi dài nhất là :")
print(test(strs))

Chuỗi gốc:
['cat', 'car', 'fear', 'center']
Chuỗi dài nhất là :
center


**Câu 198:** Viết hàm kiểm tra mảng có sắp xếp theo thứ tự tăng dần hay không nếu có trả về True không trả về False.

In [53]:
def TestSort(arr):
    temp=arr[:]
    arr.sort()
    return arr==temp
arr_A=[3,5,9,10,29]
TestSort(arr_A)

True

**Câu 199:** Viết hàm kiểm tra danh sách sao cho tổng hai số bằng 0 kết quả trả về vị trí 2 index đó.

<div style="padding: 15px; border: 2px solid transparent; border-color: transparent; margin-bottom: 5px; margin-top: 10px;border-radius: 2px; color: #000000; background-color:#A4D3EE; border-color: #FFD39B;Width:60%">

Example: [1, -4, 6, 7, 4]
<br>
<br>
Số 1 có thể cộng với [-4, 6, 7, 4] bằng [-3, 7, 8, 5]
<br>
<br>
Số -4 có thể cộng với [6, 7, 4] bằng [2, 3, 0]
<br>
<br>
Số 6 có thể cộng với [7, 4] bằng [13, 10]
<br>
<br>
Số 7 có thể cộng với [4] bằng [11]
<br>
<br>
Input: [1, -4, 6, 7, 4]
<br>
<br>
Output: [4, 1]

</div>

In [54]:
def test(nums):
    s = set(nums)
    for i in s:
        if -i in s:
            return [nums.index(i), nums.index(-i)]

nums = [1, -4, 6, 7, 4]
print("Danh sách gốc:")
print(nums) 
print("Index các số cộng với nhau bằng 0:")
print(test(nums))

Danh sách gốc:
[1, -4, 6, 7, 4]
Index các số cộng với nhau bằng 0:
[4, 1]


-----